stellar-contract-env-guest/src/guest.rs

```diff
@@ -4,11 +4,60 @@
 use stellar_contract_env_common::call_macro_with_all_host_functions;
 
 use super::{Env, EnvBase, Object, RawVal, Symbol};
+#[cfg(target_family = "wasm")]
+use static_assertions as sa;
 
 // In guest code the environment is global/implicit, so is represented as a unit struct.
 #[derive(Copy, Clone, Default)]
 pub struct Guest;
 
+// The Guest struct is only meaningful when compiling for the WASM target. All
+// these fns should not be called at all because the SDK's choice of Env should be
+// Host for a non-WASM build.
+#[cfg(not(target_family = "wasm"))]
+impl EnvBase for Guest {
```

Comment:
❗ We only build the guest crate as a whole for wasm. When will this crate be used in non-wasm?

Reply:
Should never be. This is a just-in-case. If you like I can remove it.

Reply:
(Note there are static asserts that we're on a 32-bit platform in the other case, so like the crate won't build for most hosts anyways, I don't know if maybe we want it to build-but-be-inert in some other case?)

Reply:
I think it's fine to remove this. The assumption is the entire crate is wasm32 only. Even in the SDK we never import it for non-wasm builds.

Reply:
Oh, although maybe the crate builds for non-wasm in local builds of this workspace. I don't mind if it's included.